### PR TITLE
systemd_units: fix parsing

### PR DESCRIPTION
**Pull request by Mic92**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

**Comment by Mic92** [comment]
It fixes this error:

```
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: iscsi.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: modprobe@drm.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: nfs-server.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: plymouth-quit-wait.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: plymouth-quit.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: plymouth-start.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: rbdmap.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: syslog.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: systemd-hwdb-update.service
Nov 10 10:05:40 rock telegraf[20327]: 2020-11-10T09:05:40Z E! [inputs.systemd_units] Error in plugin: Error parsing field 'load', value not in map: systemd-sysusers.service
```

File: plugins/inputs/systemd_units/systemd_units_linux.go [hunk 1]
```diff
@@ -150,6 +151,10 @@ func (s *SystemdUnits) Gather(acc telegraf.Accumulator) error {
 			acc.AddError(fmt.Errorf("Error parsing line (expected at least 4 fields): %s", line))
 			continue
 		}
+		// not found or failed units might have a leading asterisk
+		if data[0] == "*" {
+			data = data[1:]
+		}
```
Comment (srebhan) [inline plugins/inputs/systemd_units/systemd_units_linux.go:157]: The problem is that we remove things _after_ the length check. There might be cases where you get four fields, but an asterix in the beginning and then end up with only three fields left in _data_.

How about adding
```suggestion
if strings.HasPrefix(line, "*") {
     line = strings.replace(line, "*", "", 1)
}
```
above, directly after the assignment to line (line 148 I think) and **before** creating _data_?

**Review by srebhan — CHANGES_REQUESTED** [review]

**Comment by srebhan** [comment]
Does this also relate to issue #7990?

**Review by ssoroka — APPROVED** [review]

**Review by srebhan — APPROVED** [review]

**Comment by ssoroka** [comment]
this might be resolved now as of #7991

**Comment by Mic92** [comment]
Yes.

**Pull request closed without merging**